packages/admin-rte/src/core/Controls/ControlButton.tsx

```diff
@@ -14,12 +15,68 @@ export interface IProps {
     Icon?: (props: SvgIconProps) => JSX.Element;
 }
 
-export default function ControlButton({ disabled = false, selected = false, onButtonClick, icon, children, Icon: deprecatedIcon }: IProps) {
+function ControlButton({
+    classes,
+    disabled = false,
+    selected = false,
+    onButtonClick,
+    icon,
+    children,
+    Icon: deprecatedIcon,
+}: IProps & WithStyles<typeof styles>) {
     const Icon = icon || deprecatedIcon;
+
+    const rootClasses: string[] = [classes.root];
+    if (selected) rootClasses.push(classes.selected);
+    if (disabled) rootClasses.push(classes.disabled);
```

Comment:
You could also use the disabled selector https://developer.mozilla.org/en-US/docs/Web/CSS/:disabled for applying styles to a disabled element, instead of applying a disabled-class which seems redundant to me. Is there any advantage doing it like this?

Reply:
Yes, good point 👍🏻